tests/core/test_examples.py

```diff
@@ -35,46 +35,172 @@
         stories,
         os.path.join(p, "models", "dialogue"),
         endpoints=endpoints,
-        policy_config="rasa/cli/default_config.yml",
+        policy_config="examples/formbot/config.yml",
     )
-    response = {
-        "events": [
-            {"event": "form", "name": "restaurant_form", "timestamp": None},
-            {
-                "event": "slot",
-                "timestamp": None,
-                "name": "requested_slot",
-                "value": "cuisine",
-            },
-        ],
-        "responses": [{"template": "utter_ask_cuisine"}],
-    }
+
+    def response_slot(slot):
```

Comment:
Isn't there a simpler and more concise way to test it? Mocking the action server seems like a lot of pain and makes it hard to understand which specific functionality is actually tested here.

Something like (you can use `monkeyPatch` to fake the predictions of the policies, so that you can test exactly the cases you want to cover):

```

def test...(...):
  test_tracker = ...
  domain = ...
  predictions = policy_ensemble._best_policy_prediction(tracker, domain)

  assert ...
```

Reply:
I don't know, I just extended existing test to full story, to test `restart`. I don't know the other way to test it